packages/flutter/test/material/dropdown_test.dart

```diff
@@ -3681,4 +3685,58 @@ void main() {
         ..rrect(rrect: const RRect.fromLTRBXY(0.0, 0.0, 800.0, 208.0, radius, radius)),
     );
   });
+
+  testWidgets('DropdownButton hint alignment', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+    const String hintText = 'hint';
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 348.0);
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      alignment: AlignmentDirectional.center,
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 356.0);
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      alignment: AlignmentDirectional.centerEnd,
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 364.0);
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      isExpanded: true,
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 0.0);
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      isExpanded: true,
+      alignment: AlignmentDirectional.center,
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 356.0);
+
+    await tester.pumpWidget(buildFrame(
+      buttonKey: buttonKey,
+      mediaSize: const Size(800, 600),
+      isExpanded: true,
+      alignment: AlignmentDirectional.centerEnd,
+      hint: const Text(hintText)),
+    );
+    expect(tester.getTopLeft(find.text(hintText)).dx, 712.0);
+  });
```

Comment:
LGTM. Just double check, do we need tests to cover AlightmentDirectional.topStart, AlightmentDirectional.topEnd, AlightmentDirectional.bottomStart, AlightmentDirectional.buttomEnd and etc.?

Reply:
Hey, thanks for looking into this. It makes sense it is certainly possible to align at corners when setting `itemHeight` 
I've updated the test and added comments to make it easier to follow the large test.